src/overlays/actors/ovl_En_Door_Etc/z_en_door_etc.c

```diff
@@ -27,42 +32,219 @@ const ActorInit En_Door_Etc_InitVars = {
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 100, 40, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AC2634[] = {
+typedef struct {
+    /* 0x0 */ s16 sceneNum;
+    /* 0x2 */ u8 dListIndex;
+    /* 0x4 */ s16 objectId;
+} EnDoorEtcInfo; // size: 0x6;
+
+EnDoorEtcInfo sObjInfo[] = {
+    { SCENE_MITURIN, 1, OBJECT_NUMA_OBJ },
+    { -1, 0, GAMEPLAY_KEEP },
+    { -1, 0xD, GAMEPLAY_FIELD_KEEP },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+    { 0, 0, OBJECT_UNSET_0 },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_U16(shape.rot.x, 0, ICHAIN_CONTINUE),
     ICHAIN_U16(shape.rot.z, 0, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80AC25A0;
-extern InitChainEntry D_80AC2634[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2044.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC20A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2118.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2154.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC21A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2354.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC24A8.s")
+void EnDoorEtc_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    s32 objectIndex;
+    EnDoorEtcInfo* objectInfo = sObjInfo;
+    s32 i;
+    EnDoorEtc* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.shape.rot.x = -0x4000;
+    this->angle = 0;
+    for (i = 0; i < 0xF; i++, objectInfo++) {
```

Comment:
```suggestion
    for (i = 0; i < 15; i++, objectInfo++) {
```
Iterations can be in decimal. Too bad it doesn't match `ARRAY_COUNT(sObjInfo)`